File: solutions/ruby/space-age/2/space_age.rb
class SpaceAge
  EARTH_YEAR_SECS = 31557600.to_f
  ORBITAL_PERIODS = {
    'earth': 1, 'mercury': 0.2408467,
    'venus': 0.61519726, 'mars': 1.8808158,
    'jupiter': 11.862615, 'saturn': 29.447498,
    'uranus':  84.016846, 'neptune': 164.79132
  }

  def initialize(seconds)
    @earth_year = seconds / EARTH_YEAR_SECS
  end

  ORBITAL_PERIODS.each do |planet, period|
    define_method("on_#{planet}") do
      @earth_year / period
    end
  end

end

File: solutions/go/space-age/1/space-age.go
package space

type Planet string
var earthYearSecond float64 = 31557600

func Age(second float64, planet Planet) float64 {
	var space_year float64
	switch planet {
	case "Earth":
		space_year=  1
	case "Mercury":
		space_year = 0.2408467
	case "Venus":
		space_year = 0.61519726
	case "Mars":
		space_year = 1.8808158
	case "Jupiter":
		space_year = 11.862615
	case "Saturn":
		space_year = 29.447498
	case "Uranus":
		space_year = 84.016846
	case "Neptune":
		space_year = 164.79132
	}
	second = second / earthYearSecond
	return   second / space_year
}